### PR TITLE
Input failure loop fix

### PR DESCRIPTION
**Pull request by jmagly**
So there are a few issues here that going into a larger design discussion so for now I re-worked the prompts, added a bit more option there and created a fallback output if the LLM decides to break format. 

This also removes the LLM cycle where it tries to fix its own output. Def would love to play with that again for sure. The hardcoded "you are naughty dont break format again" should do the trick for now. 

Learned something doing this one, you cant give it an ounce of an idea a human is at the console, it just overrides the world. You cant even use words like "human", breaks it right out of the box.

### CODE REVIEW

**Comment by Torantulino** [comment]
Okay great!

We're currently setting up a benchmark system for testing the performance of changes like this, so we ensure we're always moving in the right direction.

**Comment by jmagly** [comment]
dont worry about it for now, still not really in the best spot

**Comment by jmanhype** [comment]
im getting this error still HELP 

SYSTEM:  Command browse_website returned: Error: 'Config' object has no attribute 'user_agent_header'

**Comment by jmanhype** [comment]
import os
import openai
from dotenv import load_dotenv
# Load environment variables from .env file
load_dotenv()

class Singleton(type):
    """
    Singleton metaclass for ensuring only import os
import openai
from dotenv import load_dotenv
# Load environment variables from .env file
load_dotenv()

class Singleton(type):
    """
    #Singleton metaclass for ensuring only one instance of a class.
    """

    _instances = {}

    def __call__(cls, *args, **kwargs):
        if cls not in cls._instances:
            cls._instances[cls] = super(
                Singleton, cls).__call__(
                *args, **kwargs)
        return cls._instances[cls]


class Config(metaclass=Singleton):
    """
    #Configuration class to store the state of bools for different scripts access.
    """

    def __init__(self):
        self.continuous_mode = False
        self.speak_mode = False
        # TODO - make these models be self-contained, using langchain, so we can configure them once and call it good 
        self.fast_llm_model = os.getenv("FAST_LLM_MODEL", "gpt-3.5-turbo") 
        self.smart_llm_model = os.getenv("SMART_LLM_MODEL", "gpt-4")
        self.fast_token_limit = int(os.getenv("FAST_TOKEN_LIMIT", 4000))
        self.smart_token_limit = int(os.getenv("SMART_TOKEN_LIMIT", 8000))
        
        self.openai_api_key = os.getenv("OPENAI_API_KEY")
        self.elevenlabs_api_key = os.getenv("ELEVENLABS_API_KEY")
        
        self.google_api_key = os.getenv("GOOGLE_API_KEY")
        self.custom_search_engine_id = os.getenv("CUSTOM_SEARCH_ENGINE_ID")

        # User agent headers to use when browsing web
        # Some websites might just completely deny request with an error code if no user agent was found.
        self.user_agent_header = {"User-Agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36"}


        # Initialize the OpenAI API client
        openai.api_key = self.openai_api_key

    def set_continuous_mode(self, value: bool):
        self.continuous_mode = value

    def set_speak_mode(self, value: bool):
        self.speak_mode = value

    def set_fast_llm_model(self, value: str):
        self.fast_llm_model = value

    def set_smart_llm_model(self, value: str):
        self.smart_llm_model = value

    def set_fast_token_limit(self, value: int):
        self.fast_token_limit = value

    def set_smart_token_limit(self, value: int):
        self.smart_token_limit = value

    def set_openai_api_key(self, value: str):
        self.openai_api_key = value
    
    def set_elevenlabs_api_key(self, value: str):
        self.elevenlabs_api_key = value
        
    def set_google_api_key(self, value: str):
        self.google_api_key = value
    
    def set_custom_search_engine_id(self, value: str):
        self.custom_search_engine_id = value instance of a class.
    """

    _instances = {}

    def __call__(cls, *args, **kwargs):
        if cls not in cls._instances:
            cls._instances[cls] = super(
                Singleton, cls).__call__(
                *args, **kwargs)
        return cls._instances[cls]


class Config(metaclass=Singleton):
    """
    Configuration class to store the state of bools for different scripts access.
    """

    def __init__(self):
        self.continuous_mode = False
        self.speak_mode = False
        # TODO - make these models be self-contained, using langchain, so we can configure them once and call it good 
        self.fast_llm_model = os.getenv("FAST_LLM_MODEL", "gpt-3.5-turbo") 
        self.smart_llm_model = os.getenv("SMART_LLM_MODEL", "gpt-4")
        self.fast_token_limit = int(os.getenv("FAST_TOKEN_LIMIT", 4000))
        self.smart_token_limit = int(os.getenv("SMART_TOKEN_LIMIT", 8000))
        
        self.openai_api_key = os.getenv("OPENAI_API_KEY")
        self.elevenlabs_api_key = os.getenv("ELEVENLABS_API_KEY")
        
        self.google_api_key = os.getenv("GOOGLE_API_KEY")
        self.custom_search_engine_id = os.getenv("CUSTOM_SEARCH_ENGINE_ID")

        # Initialize the OpenAI API client
        openai.api_key = self.openai_api_key

    def set_continuous_mode(self, value: bool):
        self.continuous_mode = value

    def set_speak_mode(self, value: bool):
        self.speak_mode = value

    def set_fast_llm_model(self, value: str):
        self.fast_llm_model = value

    def set_smart_llm_model(self, value: str):
        self.smart_llm_model = value

    def set_fast_token_limit(self, value: int):
        self.fast_token_limit = value

    def set_smart_token_limit(self, value: int):
        self.smart_token_limit = value

    def set_openai_api_key(self, value: str):
        self.openai_api_key = value
    
    def set_elevenlabs_api_key(self, value: str):
        self.elevenlabs_api_key = value
        
    def set_google_api_key(self, value: str):
        self.google_api_key = value
    
    def set_custom_search_engine_id(self, value: str):
        self.custom_search_engine_id = value

File: .gitignore
Comment (nponeccop) [inline .gitignore:1]: Is this an unrelated change? Rollback

File: scripts/chat.py [hunk 1]
```diff
@@ -35,17 +35,22 @@ def chat_with_ai(
...
-            str: The AI's response.
+prompt: Rules for the AI.
+user_input: User's input.
+full_message_history: All user-AI messages.
+permanent_memory: AI's permanent memory items.
+token_limit: Maximum tokens in API call.
+
+Returns:
+
+str: AI's response.
```
Comment (nponeccop) [inline scripts/chat.py:53]: Why the alignment? Doesn't """" have provisions for such messages?

**Review by nponeccop — CHANGES_REQUESTED** [review]
Overall the scope is just too big. But removing .gitignore change is usually a must for it to merge. Submit your .gitignore separately. And resolve the conflicts. Thanks to the big scope the conflicts are expected to be terrible.

It says it just fixes the loop, but it adds a dozen of unrelated things, such as prompt templating and whatnot. Split.

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by merwanehamadi** [comment]
@jmagly   we're currently building challenges in order to measure the effect of these types of changes, and I would like to discuss with you.

Please join use on Discord through this link https://discord.gg/autogpt (if not already)

DM me on the Auto-GPT discord channel (my discord is merwanehamadi).

**Comment by Boostrix** [comment]
my 2c: I would suggest to team up with other folks who are working on overlapping efforts and come up with a common infrastructure to easily have several prompt configs side by side, as per: https://github.com/Significant-Gravitas/Auto-GPT/pull/1874#issuecomment-1537191222

**Comment by Pwuts** [comment]
This is an interesting PR, but unfortunately it has fallen out of date and the merge conflicts have become too complex to fix. You're welcome to join us on Discord if you are still interested to contribute!

**Pull request closed without merging**